src/charm.py

```diff
@@ -183,32 +186,33 @@ def _on_insert_key_action(self, event: ops.ActionEvent) -> None:
         if not re.match(keypattern, address):
             event.fail("Illegal key pattern, did your public key/address start with 0x ?")
         else:
-            rpc_port = ServiceArgs(self._stored.service_args).rpc_port
+            rpc_port = ServiceArgs(self._stored.service_args, self._stored.relay_rpc_urls).rpc_port
             PolkadotRpcWrapper(rpc_port).insert_key(mnemonic, address)
 
     def _on_restart_node_service_action(self, event: ops.ActionEvent) -> None:
         utils.restart_service()
         if not utils.service_started():
             event.fail("Could not restart service")
-        self.unit.status = ops.ActiveStatus("Node service restarted")
+        self.update_status()
```

Comment:
set_results sound like a good add to this. I'm thinking like this:
The feedback from the action tells the operator if the service was successfully restarted. This is more on OS level and does not take the blockchain software in consideration.
The update_status goes a bit further and gives feedback to the operator if the blockchain software is actually running and healthy.

What happened to me, and why I made this change:
The service started up -> status "Node service restarted" nice :+1: 
Now it can take several minutes before the update_status hook is triggered. Once it does it tells me that the service is not running. WTF?
The actions successfully restarted the service but shortly thereafter it stopped because the blockchain software stopped due to wrong service arguments.

Reply:
I have added set_results to the actions now.